packages/compass-collection/src/components/collection/collection.tsx

```diff
@@ -138,6 +181,14 @@ const Collection: React.FunctionComponent<CollectionProps> = ({
     };
   }, [localAppRegistry, onSubTabClicked, tabs]);
 
+  const showInsights = usePreference('showInsights', React);
+
+  const insights = useMemo(() => {
+    return showInsights && pipeline?.length
+      ? getInsightsForPipeline(pipeline, isAtlas)
+      : [];
+  }, [pipeline, isAtlas, showInsights]);
```

Comment:
`isAtlas` seems like the only prop we are currently not passing to collection header, maybe makes sense to just pass it and move this whole logic there?

Reply:
Thanks. I addressed this in [6a0050a](https://github.com/mongodb-js/compass/pull/4564/commits/6a0050a112d4d247ee15bd6b5dfb26e4e47ff9e8)